test/bio_memleak_test.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include <stdio.h>
+#include <string.h>
+#include <openssl/buffer.h>
+#include <openssl/bio.h>
+
+#include "testutil.h"
+
+static int test_bio_memleak(void)
+{
+    int ok = 0;
+    BIO *bio;
+    BUF_MEM bufmem;
+    const char *str = "BIO test\n";
+    char buf[100];
+
+    bio = BIO_new(BIO_s_mem());
+    if (bio == NULL)
+        goto finish;
+    bufmem.length = strlen(str) + 1;
+    bufmem.data = (char *) str;
+    bufmem.max = bufmem.length;
+    BIO_set_mem_buf(bio, &bufmem, BIO_NOCLOSE);
+    BIO_set_flags(bio, BIO_FLAGS_MEM_RDONLY);
+
+    if (BIO_read(bio, buf, sizeof(buf)) <= 0)
+	goto finish;
+
+    if (strcmp(buf, str) != 0)
```

Comment:
style: *IMHO, I will write this more shortly:
`     ok = strcmp(buf, str) == 0;`

Reply:
Is this official openssl style?  I personally think that what you suggest is harder to read, but I know opinions on this vary.  I'm not sure who is official and who makes the decisions, but I can change this if necessary.

Reply:
I guess to be more clear, what the way I wrote it is more consistent with the rest of the function, and that's why I preferred that way.  Though "finish" might be better named "err" or something like that.

Reply:
Either form would be acceptable according to OpenSSL style